kernel/cClosure.ml

```diff
@@ -85,8 +85,8 @@ and fterm =
   | FAtom of constr (* Metas and Sorts *)
   | FFlex of table_key
   | FInd of pinductive
-  | FConstruct of pconstructor
-  | FApp of fconstr * fconstr array
+  | FConstruct of pconstructor * fconstr array
+  | FApp of fconstr * fconstr array (* head must not be fconstruct *)
```

Comment:
How can you enforce that given then zupdate mutability?

Reply:
not sure if it's important to enforce TBH